packages/design-system/src/components/Dialog/Dialog.tsx

```diff
@@ -1,49 +1,170 @@
+  /**
+   * Aria label for the close button
+   */
+  ariaCloseLabel?: string;
+  /**
+   * The modal's body content
+   */
+  children: React.ReactNode;
+  /**
+   * Additional classes to be added to the root dialog element.
+   */
+  className?: string;
+  /**
+   * Size of the close button. See [Button component]({{root}}/components/button/#components.button.react)
+   */
+  closeButtonSize?: DialogCloseButtonSize;
+  /**
+   * For internationalization purposes, the text for the "Close" button must be
+   * passed in as a prop.
+   */
+  closeButtonText?: React.ReactNode;
+  /**
+   * Variation string to be applied to close button component. See [Button component]({{root}}/components/button/#components.button.react)
+   */
+  closeButtonVariation?: ButtonVariation;
+  /**
+   * The icon to display as part of the close button
+   */
+  closeIcon?: React.ReactNode;
+  /**
+   * @hide-prop [Deprecated] This prop has been renamed to `closeButtonText`
+   * @hide-prop The text for the "Close" button
+   */
+  closeText?: React.ReactNode;
+  /**
+   * Enable exiting the dialog when a user presses the Escape key.
+   * [Read more on react-aria-modal docs.](https://github.com/davidtheclark/react-aria-modal#escapeexits)
+   */
+  escapeExits?: boolean;
+  /**
+   * @hide-prop [Deprecated] This prop has been renamed to `escapeExits`.
+   * @hide-prop Disable exiting the dialog when a user presses the Escape key.
+   */
+  escapeExitDisabled?: boolean;
+  /**
+   * Same as `applicationNode`, but a function that returns the node instead of
+   * the node itself. The function will not be called until after the component
+   * mounts, so it's safe to use browser globals and refer to DOM nodes within
+   * it (e.g. `document.getElementById(..)`)
+   */
+  getApplicationNode?: (...args: any[]) => any;
+  /**
+   * Additional classes to be added to the header, which wraps the heading and
+   * close button.
+   */
+  headerClassName?: string;
+  /**
+   * The Dialog's heading, to be rendered in the header alongside the close button.
+   */
+  heading?: React.ReactNode;
+  /**
+   * Set focus to a specific element that should receive initial focus (if `focusDialog={false}`).
+   * [Read more on react-aria-modal docs.](https://github.com/davidtheclark/react-aria-modal#initialfocus)
+   */
+  initialFocus?: string;
+  /**
+   * A method to handle the state change of exiting (or deactivating)
+   * the modal. It will be invoked when the user presses Escape, or clicks outside
+   * the dialog (if `underlayClickExits=true`).
+   */
+  onExit?: (...args: any[]) => any;
+  size?: DialogSize;
+  /**
+   * @hide-prop [Deprecated] This prop has been renamed to `heading`.
+   */
+  title?: React.ReactNode;
+  /**
+   * Enable exiting the dialog when a user clicks the underlay.
+   */
+  underlayClickExits?: boolean;
+  /**
+   * Allow additional AriaModal props to be passed to Dialog
+   */
+  // eslint-disable-next-line camelcase
+  [additional_props: string]: unknown;
```

Comment:
The reason I want to look at the type definitions for `react-aria-modal` is that I wonder if we can extend the props for that component here rather than having this generic string index with an unknown type.

Reply:
Gotcha, those are in `node_modules/react-aria-modal/src/react-aria-modal.js`

`
     7  static defaultProps = {
     8     underlayProps: {},
     9     dialogId: 'react-aria-modal-dialog',
    10     underlayClickExits: true,
    11     escapeExits: true,
    12     underlayColor: 'rgba(0,0,0,0.5)',
    13     includeDefaultStyles: true,
    14     focusTrapPaused: false,
    15     scrollDisabled: true
    16   };
`

plus the others listed there, `titleText`, `titleId`, `getApplicationNode`, `underlayProps`, `dialogStyle`

Reply:
No static definitions but yes I could create an interface for those

Reply:
There are some types here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-aria-modal/index.d.ts we could include I suppose? Though they aren't a part of the core package it adds a dependency..

Reply:
What I don't understand is how TypeScript is even allowing us to use this without type definitions. Normally it yells at us. If we add `@types/react-aria-modal` like you linked to, we could extend your props with `AriaModalProps`, and that would give us everything we need.

Reply:
That library contains a ton of definitions I don't think we want to mess with, I've extracted the type from the set and extended it in the dialog component here.. I actually tried to install that library and ran into a bunch of version issues with yarn it wouldn't let me do it without a struggle.. Also if you look at that libraries issues list it's like 500+ long, a little spooky.. let me know what you think

Reply:
Yikes! In that case TypeScript mysteriously ignoring it is a blessing. 👀 I guess we should just ignore it too. Recreating the props type definition sounds good to me 👍